crypto/src/hash/types.rs

```diff
@@ -31,10 +30,24 @@ impl From<HashBytes> for BaseHash {
     }
 }
 
+impl<'a> From<&'a [u8]> for BaseHash {
```

Comment:
Do we still need this impl, after implementing `AsRef` below?

Reply:
I think the other hash implementations still need it, for example this doesn't work if there's no `AsRef` for `BaseHash`:
```
pub struct StateHash(BaseHash);

impl AsRef<[u8]> for StateHash {
    fn as_ref(&self) -> &[u8] {
        &self.0.as_ref()
    }
}
```